modules/fedimint-ln-server/src/lib.rs

```diff
@@ -256,10 +251,6 @@ impl ServerModuleInit for LightningInit {
     }
 
     async fn init(&self, args: &ServerModuleInitArgs<Self>) -> anyhow::Result<DynServerModule> {
-        // Ensure all metrics are initialized
```

Comment:
Removing this will only show the metric if the value is greater than zero. I think it should show even if it's zero.

Reply:
AFAIK Prometheus is fine with that, and for most purposes it just doesn't matter. Prometheus (or it's formulas) will treat missing values as zero, and will handle it just like it handles e.g. counter resets.

If it was necessary, it would be happening automatically and user libraries wouldn't require users to do second double accounting of all values.

Pretty much only very rarely updated gauges need manually initializing on start.

Reply:
> If it was necessary, it would be happening automatically and user libraries wouldn't require users to do second double accounting of all values.

It does happen automatically. The problem is that we are using lazy val here.

Reply:
Good point, but it still works fine, and such lazy_static usage is mentioned all over the examples, etc. and no one worries.

Did you experience any e2e problems with it in queries, etc.?

Reply:
The problem is the discoverability of the metrics. If you are going to remove the initialization of metrics then we will need a documentation with all the metrics so the user can create alerts and dashboards using them.

If discoverability isn't important then I wonder why you are adding a prefix to all metrics :thinking:

Reply:
These metrics will show up right away, typically after a couple of seconds the first time they are touched. As soon as prometheus sees them it will rememeber them. In a long running software like fedimintd it's completely not an issue. @douglaz

Reply:
This isn't true for something like LN_OUTPUT_OUTCOME_CANCEL_OUTGOING_CONTRACT. It may take weeks or months to happen, but when it happens you want to know right away because it may indicate an issue on the gateway.

Reply:
@douglaz Then I rather eagerly initialize these handful ones that are like that than maintain the whole list of everything.

Gauges that change very rarely (.e.g blockchain block height, session count, etc.) need initializing like that with a proper value on start as well.

Any other one you think is like that?

Reply:
> Any other existing ones you think are like that?

I guess all wallet ones are like that... :D